test/shared/src/main/scala/zio/test/Spec.scala

```diff
@@ -78,9 +95,9 @@ final case class Spec[+L, +T](caseValue: SpecCase[L, T, Spec[L, T]]) { self =>
   /**
    * Determines if all node in the spec are satisfied by the given predicate.
    */
-  final def forall(f: SpecCase[L, T, Unit] => Boolean): Boolean =
-    fold[Boolean] {
-      case c @ SuiteCase(_, specs, _) => specs.forall(identity) && f(c.map(_ => ()))
+  final def forall[R1 <: R, E1 >: E](f: SpecCase[R, E, L, T, Any] => ZIO[R1, E1, Boolean]): ZIO[R1, E1, Boolean] =
+    fold[ZIO[R1, E1, Boolean]] {
+      case c @ SuiteCase(_, specs, _) => specs.flatMap(ZIO.collectAll(_).map(_.forall(identity))).zipWith(f(c))(_ && _)
```

Comment:
Sequential specialization. Possibly look at the strategy in the node?

Reply:
We could. The only time this method is used in the code base today is on executed specs. This goes a little to some of the other points about the value of these methods on specs that haven't been executed now that the suites are effectual.